### DISCUSSION TITLE
release-notes in postupgrade templates

### DISCUSSION THREAD

**Discussion by JensAc** [renovatebot/renovate#20451] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.141.0

### If you're self-hosting Renovate, select which platform you are using.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitHub

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Hi there :wave: , 
I want to append the release notes which are included in the PR to an existing file in order to keep track of all changelogs.

### What you tried so far.

 I have prepared a minimal working example [here](https://github.com/JensAc/renovate-mwe) and the [open PR](https://github.com/JensAc/renovate-mwe/pull/3#issue-1587170448) contains release notes. I have 
```json
"postUpgradeTasks": {
    "commands": ["./hack/export-renovate-releasenotes.sh {{{depName}}} {{{currentVersion}}} {{{newVersion}}} {{{body}}} >> release-notes/next.md"],
    "fileFilters": ["release-notes/next.md"],
    "executionMode": "update"
  }
```
in my [renovate config](https://github.com/JensAc/renovate-mwe/blob/main/.github/renovate.json5) and would expect that the release notes are provided to my [scrip](https://github.com/JensAc/renovate-mwe/blob/main/hack/export-renovate-releasenotes.sh) as the `{{{body}}}` argument, as explained in the [renovate documentation](https://docs.renovatebot.com/templates/). However, as you can see in the PR, there are no release notes in `relase-notes/next.md` and renovate calls the postUgradetask like so:
```sh
DEBUG: Executing command (repository=jensac/renovate-mwe, baseBranch=main, branch=renovate/fluxcd-flux2-0.x)
       "dep": "fluxcd/flux2",
       "command": "./hack/export-renovate-releasenotes.sh fluxcd/flux2 v0.38.2 v0.39.0  >> release-notes/next.md"
```
which obviously misses the release notes. Is this behavior intended? Am I doing something wrong or is this a bug? I would really appreciate any kind of help.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
@viceice is it possible that the notes aren't fetched yet due to the "lazy" fetching which would wait until the ensurePr function?

**Reply by viceice** [reply]
yes, we don't check the post update commands for release notes usage.

**Reply by viceice** [reply]
I think we need to update this function?
https://github.com/renovatebot/renovate/blob/0e47a1072ca4f3fa2eb1dba6a3178d93b7be2c79/lib/workers/repository/changelog/index.ts#L25-L37

called from here:

https://github.com/renovatebot/renovate/blob/0e47a1072ca4f3fa2eb1dba6a3178d93b7be2c79/lib/workers/repository/updates/branchify.ts#L75-L89

**Reply by viceice** [reply]
i see we deprecated fetching changelogs early

**Reply by rarkins** [reply]
Maybe we could make it mandatory to configure which stage the notes are fetched at, defaulting to latest/laziest

**Reply by viceice** [reply]
sounds good

**Reply by viceice** [reply]
so make `fetchReleaseNotes` an enum with values `off`, 'branch` or `pr` ?

**Reply by rarkins** [reply]
Yes

**Reply by viceice** [reply]
@HonkingGoose can you please open a feature request?

**Reply by HonkingGoose** [reply]
Here's your feature request: 😉 

- #20476
